### PR TITLE
Fix EAN2 incorrect encoding issue

### PR DESCRIPTION
**Pull request by shinbin**
I hope this pull request finds you well.
I refer to ( https://en.wikipedia.org/wiki/EAN-2 ) and fixed the convert function of ean2.dart.
I use websites ( https://www.online-barcode.com/ ) and other methods to verify whether the ean2 code is correct.
I would appreciate your consideration for merging it into the main repository and update it to ( https://pub.dev/packages/barcode ) .

Thank you for your time and consideration.

Best regards,
Andy (@shinbin)

### CODE REVIEW

File: flutter/pubspec.yaml [hunk 1]
```diff
@@ -4,14 +4,14 @@ description: >-
...
-version: 2.0.4
+version: 2.0.5
 
 environment:
   sdk: ">=2.12.0 <4.0.0"
   flutter: ">=1.16.0"
 
 dependencies:
-  barcode: ^2.2.4
+  barcode: ^2.2.5
```
Comment (DavBfr) [inline flutter/pubspec.yaml]: You don't need to change this file. Pub will take the latest version

File: image/pubspec.yaml [hunk 2]
```diff
@@ -6,14 +6,14 @@ _author: David PHAM-VAN <dev.nfet.net@gmail.com>
...
-version: 2.0.3
+version: 2.0.4
 
 environment:
   sdk: ">=2.12.0 <4.0.0"
 
 dependencies:
   args: ^2.3.0
-  barcode: ^2.2.4
+  barcode: ^2.2.5
```
Comment (DavBfr) [inline image/pubspec.yaml]: You don't need to change this file. Pub will take the latest version

File: barcode/example/main.dart [hunk 3]
```diff
@@ -82,7 +82,7 @@ void main() {
 
   buildBarcode(
     Barcode.ean2(),
-    '05',
+    '12',
```
Comment (DavBfr) [inline barcode/example/main.dart]: A unit test would be better than changing this

**Review by DavBfr — CHANGES_REQUESTED** [review]

**Comment by shinbin** [comment]
@DavBfr Thanks you for your replies. I've rollbacked some of the changes you suggested.

File: flutter/pubspec.yaml (hunk 1 above)
Comment (shinbin) [reply flutter/pubspec.yaml]: I have removed this part of the edit in [b286841](https://github.com/DavBfr/dart_barcode/pull/65/commits/b2868410d5f09fdc5ec53684aa63439cdd4d29b7) .

File: image/pubspec.yaml (hunk 2 above)
Comment (shinbin) [reply image/pubspec.yaml]: I have removed this part of the edit in [b286841](https://github.com/DavBfr/dart_barcode/pull/65/commits/b2868410d5f09fdc5ec53684aa63439cdd4d29b7) .

File: barcode/example/main.dart (hunk 3 above)
Comment (shinbin) [reply barcode/example/main.dart]: I have removed this part of the edit in [b286841](https://github.com/DavBfr/dart_barcode/pull/65/commits/b2868410d5f09fdc5ec53684aa63439cdd4d29b7).
However, the problem with this fix is the barcode drawing and has nothing to do with data conversion. I have not learned to write unit tests for graphical interfaces.

Comment (shinbin) [reply barcode/example/main.dart]: Sorry, I have known how to test barcode graphics from other unit test. 
I will add this unit test next week.
 Thanks.

**Comment by DavBfr** [comment]
Thanks!

**Pull request merged**